src/runtime/HalideRuntime.h

```diff
@@ -89,13 +89,25 @@ struct halide_mutex {
     uint64_t _private[8];
```

Comment:
Since we're starting to use C++ under #ifdef here, do we want to add default constructors that zero the memory?